packages/deployer/subtasks/deploy-contracts.js

```diff
@@ -0,0 +1,130 @@
+
+    // Wait for the transaction to finish
+    const { gasUsed, status } = await _waitForTransaction(transaction);
+
+    hre.deployer.data.transactions[transaction.hash].status = status;
+
+    const totalGasUsed = hre.ethers.BigNumber.from(hre.deployer.data.properties.totalGasUsed)
+      .add(gasUsed)
+      .toString();
+    hre.deployer.data.properties.totalGasUsed = totalGasUsed;
+
+    contractData.deployedAddress = contract.address;
+    contractData.deployTransaction = transaction.hash;
+    contractData.bytecodeHash = sourceBytecodeHash;
+  }
+});
+
+/**
+ * Prompts the user to confirm the give deployments if necessary
+ */
+async function _confirmDeployments({ toSkip, toUpdate, toCreate }) {
+  if (toSkip.length > 0) {
+    logger.info(`There are ${toSkip.length} contracts that are already up-to-date:`);
+    toSkip.forEach(([source]) => logger.notice(`  ${source}`));
+  }
+
+  if (toUpdate.length > 0) {
+    logger.info(`The following ${toUpdate.length} contracts are going to be updated:`);
+    toUpdate.forEach(([source]) => logger.notice(`  ${source}`));
+  }
+
+  if (toCreate.length > 0) {
+    logger.info(
+      `The following ${toCreate.length} contracts are going to be deployed for the first time:`
+    );
+    toCreate.forEach(([source]) => logger.notice(`  ${source}`));
+  }
+
+  await prompter.confirmAction('Are you sure you want to make these changes?');
+}
+
+/**
+ * Sort contracts by the ones that doesn't need deployment, the ones that are going
+ * to be re-deployed with updated code, and the ones that are going to be deployed
+ * for the first time.
+ */
+async function _processContracts(contracts) {
+  const toSkip = [];
+  const toUpdate = [];
+  const toCreate = [];
+
+  for (const [contractPath, contractData] of Object.entries(contracts)) {
+    if (hre.network.name === 'hardhat' || !contractData.deployedAddress) {
+      toCreate.push([contractPath, contractData]);
+    } else {
+      const sourceBytecodeHash = getContractBytecodeHash(contractPath);
+      const remoteBytecodeHash = await getAddressBytecodeHash(contractData.deployedAddress);
+
+      if (sourceBytecodeHash === remoteBytecodeHash) {
+        toSkip.push([contractPath, contractData]);
+      } else {
+        toUpdate.push([contractPath, contractData]);
+      }
+    }
+  }
+
+  return { toSkip, toUpdate, toCreate };
+}
+
+/**
+ * Deploy the given contract using ethers.js
+ */
+async function _createAndDeployContract(contractName) {
+  logger.success(`Deploying ${contractName}`);
+
+  const factory = await hre.ethers.getContractFactory(contractName);
+  const contract = await factory.deploy();
+
+  if (!contract.address) {
+    throw new Error(`Error deploying ${contractName}`);
+  }
+
+  return { contract, transaction: contract.deployTransaction };
+}
+
+/**
+ * Given a transaction, wait for it to finish and return the state with the gas used.
+ */
+async function _waitForTransaction(transaction) {
```

Comment:
We might want to move this out to a util later, when some other sub-task needs it. Eventually, it might even move out to a core-js util outside of the package. Or maybe not. We'll see. It's good here for now.

Reply:
For sure! Would be handy to have a single function (or prototype) that does `_deployContract` + `_waitForTransaction`